src/lib.rs

```diff
@@ -25,61 +24,54 @@ use crate::ip_who::IpWhoIs;
 use crate::us_ofac::validate_country_code;
 
 fn get_providers() -> Vec<Box<dyn LocationProvider>> {
-  let mut providers: Vec<Box<dyn LocationProvider>> = vec![];
+    let mut providers: Vec<Box<dyn LocationProvider>> = vec![];
 
-  #[cfg(feature = "ip_info_provider")]
-  providers.push(Box::new(IpInfoIoFetch {}));
+    #[cfg(feature = "ip_info_provider")]
+    providers.push(Box::new(IpInfoIoFetch {}));
 
-  #[cfg(feature = "ip_who_provider")]
-  providers.push(Box::new(IpWhoIs {}));
+    #[cfg(feature = "ip_who_provider")]
+    providers.push(Box::new(IpWhoIs {}));
 
-  providers
+    providers
 }
 
 // Note: rejected options
 // https://icanhazip.com/ - returns only IP, not enough
 // https://api.iplocation.net/?ip=8.8.8.8 - country only, no region
 // https://ipbase.com/ - requires free plan sign-up
 
-///
-///
-pub struct ComplianceChecker {
-  config: Option<Configuration>,
+/// Main object responsible for validating the country and region
+pub struct ComplianceChecker<'a> {
+    config: Option<&'a Configuration>,
 }
 
-impl ComplianceChecker {
-  ///
-  /// Creates an instance of the Compliance Checker
-  pub fn new(config: Option<Configuration>) -> Self {
-    Self {
-      config
-    }
-  }
-
-  /// Validates
-  pub fn validate_host(&self) -> Result<(), Error> {
-    let providers = get_providers();
-    for provider in providers {
-      match provider.location(self.config.borrow()) {
-        Ok(location) => return validate_country_code(&location),
-        _ => continue, // try next fetcher
-      }
+impl<'a> ComplianceChecker<'a> {
+    ///
+    /// Creates an instance of the Compliance Checker
+    pub fn new(config: Option<&'a Configuration>) -> Self {
+        Self { config }
     }
 
-    Err(Error::UnableToFetch)
-  }
+    /// Validates the host
+    pub fn validate_host(&self) -> Result<(), Error> {
+        let providers = get_providers();
+        for provider in providers {
+            match provider.location(self.config) {
```

Comment:
This is hopefully the only place that needs to deal with the `Option<T>` vs `Option<&T>`
```suggestion
            match provider.location(self.config.as_ref()) {
```